internal/terminal/table.go

```diff
@@ -0,0 +1,139 @@
+package terminal
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/fatih/color"
+)
+
+const (
+	logFieldHeaders = "headers"
+	logFieldData    = "data"
+)
+
+type table struct {
+	headers      []string
+	data         []map[string]string
+	columnWidths map[string]int
+}
+
+var (
+	// gutter is to set a standard for the break between columns for the table
+	gutter = strings.Repeat(" ", 2)
+)
+
+var (
+	tableFields = []string{logFieldData, logFieldHeaders}
+)
+
+func newTable(headers []string, data []map[string]interface{}) table {
+	var t table
+
+	if len(headers) == 0 {
+		return t
+	}
+
+	t.headers = headers
+	t.data = make([]map[string]string, len(data))
+	t.columnWidths = make(map[string]int, len(headers))
+
+	for _, header := range headers {
+		t.columnWidths[header] = len(header)
+	}
+
+	for i, row := range data {
+		r := make(map[string]string)
+		for _, header := range t.headers {
+			// convert the interface data into strings for printing
+			value := parseValue(row[header])
+			if width := len(value); width > t.columnWidths[header] {
+				t.columnWidths[header] = width
+			}
+			r[header] = value
+		}
+		t.data[i] = r
+	}
+	return t
+}
+
+func (t table) Message() (string, error) {
+	if err := t.validate(); err != nil {
+		return "", err
+	}
+	return fmt.Sprintf(`
+%s
+%s
+%s`,
+		t.headerString(), t.dividerString(), t.rowString()), nil
+}
+
+func (t table) Payload() ([]string, map[string]interface{}, error) {
+	if err := t.validate(); err != nil {
+		return nil, nil, err
+	}
+	return tableFields, map[string]interface{}{
+		logFieldHeaders: t.headers,
+		logFieldData:    t.data,
+	}, nil
+}
+
+func (t table) headerString() string {
+	cells := make([]string, len(t.headers))
+	for i, header := range t.headers {
+		cells[i] = fmt.Sprintf("%s%s",
+			color.New(color.Bold).SprintFunc()(header),
+			strings.Repeat(" ", t.columnWidths[header] - len(header)),
+		)
+	}
+	return strings.Join(cells, gutter)
+}
+
+func (t table) rowString() string {
+	cells := make([]string, len(t.data))
+	for i, dataMap := range t.data {
+		// if there's no data for that row, then ignore
+		if len(dataMap) > 0 {
+			row := make([]string, len(t.headers))
+			for j, header := range t.headers {
+				row[j] = fmt.Sprintf(
+					"%s%s",
+					dataMap[header],
+					strings.Repeat(" ", t.columnWidths[header] - len(dataMap[header])),
+				)
+			}
+			cells[i] = strings.Join(row, gutter)
+		}
+	}
+	return strings.Join(cells, "\n")
+}
+
+func (t table) dividerString() string {
+	divider := make([]string, len(t.headers))
```

Comment:
can this name be something pluralized, since its an array?  even `parts` would be fine, or `cells` to remain consistent with the other `*String` methods